spiceaidocs/docs/data-connectors/graphql.md

```diff
@@ -265,6 +265,44 @@ sql> select node from stargazers limit 1;
 +------------------+--------+---------------+
 ```
 
-:::warning[Limitations]
-- Automatic object unnesting will overwrite columns if multiple properties contain the same sub-keys.
-:::
\ No newline at end of file
+#### Unnesting Duplicate Columns
+
+By default, the Spice Runtime will error when a duplicate column is detected during unnesting.
+
+For example, this example `spicepod.yml` query would fail due to `name` fields:
+```yaml
+from: graphql:https://localhost
+name: stargazers
+params:
+  unnest_depth: 2
+  json_path: data.users
+  query: |
+    query {
+      users {
+        name
+        metadata {
+          name: email
+        }
+      }
+    }
+```
+
+To avoid this error, you should use [use aliases in your query](https://www.apollographql.com/docs/kotlin/advanced/using-aliases/) where possible. In the example above, we used an alias to induce the duplicate error in our query with `name: email`.
+
+The example below is more complex where a query returns a person, their name, and their starships' name. We've used an alias to rename `starship.name` to `starshipName` when unnesting, to avoid the duplicate column error:
```

Comment:
```suggestion
The example below is more complex where a query returns a person, their name, and their starships' name. An alias is used to rename `starship.name` to `starshipName` when unnesting, to avoid the duplicate column error:
```